core/txpool/list.go

```diff
@@ -357,14 +365,18 @@ func (l *list) Filter(costLimit *big.Int, gasLimit uint64) (types.Transactions,
 		}
 		invalids = l.txs.filter(func(tx *types.Transaction) bool { return tx.Nonce() > lowest })
 	}
+	// Reset total cost
+	l.subTotalCost(append(removed, invalids...)...)
```

Comment:
You can skip reallocating it all, by doing this instead: 
```
l.subTotalCost(removed...)
l.subTotalCost(invalids...)
```

Reply:
Yes your absolutely right